kimchi/src/circuits/polynomials/foreign_field_add/witness.rs

```diff
@@ -96,24 +107,46 @@ fn compute_subadd_values<F: PrimeField>(
         }
     });
 
-    // c1 = r2 - a2 - s*b2 + q*f2
-    let carry_mi =
+    // c = [ (a1 * 2^88 + a0) + s * (b1 * 2^88 + b0) - q * (f1 * 2^88 + f0) - (r1 * 2^88 + r0) ] / 2^176
+    //  <=>
+    // c = r2 - a2 - s*b2 + q*f2
+
+    let carry_bot: F = (bottom(&left_input[LO], &left_input[MI])
+        + bottom(&right_input[LO], &right_input[MI]) * sign
+        - bottom(&foreign_modulus[LO], &foreign_modulus[MI]) * field_overflow
+        - bottom(&result[LO], &result[MI]))
+        / (two_to_limb * two_to_limb);
```

Comment:
Once you update against master, for simplicity and `ExprOps` reusability you should replace all of these `two_to_limb` with T::two_to_limb()`.